backend/controller/auth.go

```diff
@@ -0,0 +1,76 @@
+package controller
+
+import (
+	"api/models"
+	"crypto/rand"
+	"encoding/json"
+	"github.com/google/uuid"
+	"log"
+	"net/http"
+)
+
+// https://developers.google.com/codelabs/passkey-form-autofill
+
+// https://webauthn.guide/#registration
+func RegisterRequest(w http.ResponseWriter, r *http.Request) {
+	challenge := make([]byte, 8) // maybe needs to be 16?
+	_, err := rand.Read(challenge)
+	if err != nil {
+		// TODO: handle error
+	}
+
+	options := models.WebAuthnOptions{
+		Challenge: challenge,
+		Rp: models.Rp{
+			Name: "Lynx Locks",
+		},
+		User: models.UserInfo{
+			Id:          uuid.NewString(), // TODO: get uuid token from url and fetch user id from activeTokens table
+			Name:        "john_james",
+			DisplayName: "John",
+		},
+		PubKeyCredParams: []models.PubKeyCredParams{
+			{
+				Alg:  -7,
+				Type: "public-key",
+			},
+			{
+				Alg:  -257,
+				Type: "public-key",
+			},
+		},
+		AuthenticatorSelection: models.AuthenticatorSelection{
+			AuthenticatorAttachment: "platform",
+			RequireResidentKey:      true,
+		},
+	}
+
+	errJson := json.NewEncoder(w).Encode(&options)
+	if errJson != nil {
+		http.Error(w, "500", http.StatusInternalServerError)
+		return
+	}
+}
+
+func RegisterResponse(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var registerReq models.RegisterReq
+	err := json.NewDecoder(r.Body).Decode(&registerReq)
```

Comment:
in addition, look at the post request for users on how to handle grabbing body variables, afaik what you have here won't error if you are missing some params on your input